multisig/src/lib.rs

```diff
@@ -18,29 +19,90 @@ mod setup;
 mod storage;
 mod util;
 
-pub const PERCENTAGE_TOTAL: u64 = 10_000; // precision of 2 decimals
+use token_module::ProxyTrait as _;
+
+pub const PERCENTAGE_TOTAL: u32 = 10_000; // precision of 2 decimals
 
 elrond_wasm::imports!();
 
 /// Multi-signature smart contract implementation.
 /// Acts like a wallet that needs multiple signers for any action performed.
-#[elrond_wasm_derive::contract]
+#[elrond_wasm::contract]
 pub trait Multisig:
     multisig_general::MultisigGeneralModule
     + setup::SetupModule
     + storage::StorageModule
     + util::UtilModule
 {
+    #[init]
+    fn init(
+        &self,
+        esdt_safe_sc_address: ManagedAddress,
+        multi_transfer_sc_address: ManagedAddress,
+        required_stake: BigUint,
+        slash_amount: BigUint,
+        quorum: usize,
+        #[var_args] board: ManagedVarArgs<ManagedAddress>,
+    ) -> SCResult<()> {
+        self.quorum().set(&quorum);
+
+        let mut duplicates = false;
+        let board_len = board.len();
+        self.user_mapper()
+            .get_or_create_users(board.into_iter(), |user_id, new_user| {
+                if !new_user {
+                    duplicates = true;
+                }
+                self.user_id_to_role(user_id).set(&UserRole::BoardMember);
+            });
+        require!(!duplicates, "duplicate board member");
```

Comment:
You can skip the setting of the board members when you have duplicates if you move the require! inside lthe ambda